### DISCUSSION TITLE
ESP32S3 with `LAN8740` support.

### DISCUSSION THREAD

**Discussion by ZanzyTHEbar** [espressif/arduino-esp32#9135] [Q&A]
Hey ya'll. I am spinning up a board and i am in the testing R&D phase right now. My other STM32 boards all use the `LAN8740` PHY, so i figured that i would use that in this project (familiarity). I am using the `esp32s3` for this board. 

i have a few questions to clarify some observations that i have made. 

I took a look at `framework-arduinoespressif32/libraries/Ethernet/src/ETH.cpp` I noticed that with IDF v > `4.4` all of the `LAN8720` driver stuff is marked as not selected in favour of the `KSZ8081` is there a reason for this? Can i safely simply remove the bits i don't need and select for the `LAN8740`? 

I also noticed that the `sdkconfig.h` automatically sets: 
```cpp
#define CONFIG_ETH_USE_SPI_ETHERNET 1
#define CONFIG_ETH_SPI_ETHERNET_DM9051 1
#define CONFIG_ETH_SPI_ETHERNET_W5500 1
```
I can not make changes to this, when i try, they do not compile in. Leading me to believe that this header is just for show and the real sdkconfig is pre-compiled ahead of time? If so, is there a way for me to change this? I'd like to use RMII, unless there is a reason that i shouldn't. I am not seeing anything in the datasheet for the esp32s3 that says's i shouldn't or couldn't use RMII. 

I also noticed the `.skip.esp32s3` what's the deal with this? Is this for testing and docs, or is this a skip file for compilation? I have a feeling its for internal development (CI/CD) and not compilation related - but i am curious. 

Anywho - TLDR: I am looking to setup an esp32s3 with ethernet using the Arduino framework and i'd like to get some feedback on the most idiomatic way forward. Thanks :)

**Comment by me-no-dev** [comment]
Only ESP32 has RMII interface (and the upcoming ESP32-P4), but all other chips can do only SPI Ethernet

**Reply by ZanzyTHEbar** [reply]
The S3 doesn't have an RMII interface? Why? How did i miss that ... (rhetorical question, surprised i missed that in the datasheet). I am pretty excited for the upcoming P4.

**Reply by ZanzyTHEbar** [reply]
Ill switch to the esp32 wroom-32E then. That seems to have an RMII interface. Thanks for the heads up. Anything else i should pay attention to? For example - do i need to setup a custom board, or is there a compatible board json for ethernet enabled esp32 boards?

**Reply by ZanzyTHEbar** [reply]
`/.platformio/packages/framework-arduinoespressif32/libraries/Ethernet/src/ETH.h:27:10: fatal error: WiFi.h: No such file or directory` Tried  to setup the example located under `/.platformio/packages/framework-arduinoespressif32/libraries/Ethernet/examples/ETH_LAN8720` and it can't seem to find the `WiFi.h` which doesn't make sense at all. Only the `ETH.h` header fails to find the file, i can use wifi perfectly fine as long as i don't include the `ETH.h` header.
